packages/compat/tests/resolver.test.ts

```diff
@@ -476,6 +476,20 @@ describe('compat-resolver', function () {
     let findDependencies = configure({ staticHelpers: true });
     expect(findDependencies('templates/application.hbs', `{{(this.myHelper 42)}}`)).toEqual([]);
   });
+  test('class defined component not failing if there is no arguments', function () {
+    let findDependencies = configure({ staticComponents: true, staticHelpers: true });
+    expect(findDependencies('templates/application.hbs', `{{this.myComponent}}`)).toEqual([]);
+  });
```

Comment:
```suggestion
  test('helper defined in component not failing if there is no arguments', function () {
    let findDependencies = configure({ staticComponents: true, staticHelpers: true });
    expect(findDependencies('templates/application.hbs', `{{#if (this.myHelper)}}{{/if}}`)).toEqual([]);
  });
```

Reply:
This one should already work (the subexpression hook already had the guard you are adding here)